developer-docs/graph/README.md

```diff
@@ -1,285 +1,155 @@
-
-<!-- GETTING STARTED -->
-
-## 💻 Getting Started
-
-To get a local copy up and running, follow these steps.
+## 🔍 Architecture Overview <a name="architecture-overview"></a>
 
-### Prerequisites
-
-In order to run this project you need:
+The Graph Service is built using NestJS and is divided into two main components:
 
-- [Docker](https://docs.docker.com/get-docker/)
-- [Nodejs](https://nodejs.org)
+1. **API**: Handles incoming HTTP requests
+2. **Worker**: Processes jobs that require blockchain interaction
 
-### Setup
+![Gateway Graph Service Architecture](./graph_service_arch.drawio.png)
 
-Clone this repository to your desired folder:
+## 🔑 Key Features <a name="key-features"></a>
 
-Example commands:
+- Fetch Graphs: Retrieve social connection data
+- Update Graph: Modify social connections
+- Watch Graphs: Monitor changes in social graphs
 
-```sh
-  git clone git@github.com:ProjectLibertyLabs/graph-service.git
-  cd graph-service
-```
+## 💻 Getting Started <a name="getting-started"></a>
 
-### Install
+This section will guide you through setting up the Graph Service for both quick start and local development.
 
-Install NPM Dependencies:
-
-```sh
-  npm install
-```
+### Prerequisites
 
-### Environment Variables
+Ensure you have the following installed:
+- [Node.js](https://nodejs.org)
+- [Docker](https://docs.docker.com/get-docker/)
 
-The application receives its configuration from the environment. Each method of launching the app has its own source for the environment. If you run a container image using Kubernetes, it is likely your environment injection will be configured in a Helm chart. For local Docker-based development, you may specifiy the environment or point to an environment file (the included [docker-compose.yaml](./docker-compose.yaml) can has a self-contained environment which may be edited to suit your purposes). If running natively using the script included in `package.json`, the app will use a local `.env` file.
+### Quick Start
 
-Environment files are documented [here](./ENVIRONMENT.md), and a sample environment file is provided [here](./env.template).
+1. Clone the repository and navigate to the project directory:
+   ```bash
+   git clone https://github.com/ProjectLibertyLabs/gateway.git
+   cd gateway
+   ```
 
-1. Copy the template values into the .env file.
-   ```sh
-   cp env.template .env
+2. Install dependencies:
+   ```bash
+   npm install
    ```
-2. Replace template values with values appropriate to your environment.
 
-### Usage
+3. Set up environment variables:
+   ```bash
+   cp env-files/graph.template.env .env
+   ```
+   Edit the `.env` file with appropriate values for your environment.
 
-Note: using [docker compose file](docker-compose.yaml) to start the services. This will start the services in development mode.
+4. Start all services using Docker Compose:
+   ```bash
+   docker compose up -d redis frequency graph-service-api graph-service-worker
```

Comment:
I don't think works. It looks like we've 2 instances of redis
- redis
- bull

I'm unaware why we've split them in the first place but that might be the issue

Reply:
you might also need to run the bull instance on the requested port

Reply:
Actually this seem to work the problem was that when you don't have a registered provider the worker just doesn't error out and just hangs there and that's something that I think we have a ticket for